File: Day_1/Puzzle1.js
const fs = require('fs');
const readline = require('readline');

async function readInByLine(){
    const sonar = []
    const fileStream = fs.createReadStream("sonar.txt");

    const rl = readline.createInterface({
        input: fileStream,
        crlfDelay: Infinity
    });

    for await (const line of rl){
        sonar.push(parseInt(line))
    }
    return sonar;
}


const  main = async () =>{
    const sonar = await readInByLine();

    const increasedCounter = (list) => {
        let increasedCount = 0;
        for(let i = 0; i < list.length-1; i++){
            if(list[i+1] > list[i]){
                increasedCount+= 1;
            }
        }
        return increasedCount;
    }

    const partOne = () => {
        let increasedCount = increasedCounter(sonar)
        console.log("Part One: " + increasedCount.toString())
    }


    partOne()

    const partTwo = () =>{
        slidingWindows = []
        for(let i = 0; i < sonar.length-2; i++){
            slidingWindows.push(sonar[i] + sonar[i+1] + sonar[i+2])
        }
        let increasedCount = increasedCounter(slidingWindows)
        console.log("Part Two: " + increasedCount.toString())
    }
    partTwo()
}


main()

File: Day_1/Puzzle1.cs
using System;
using System.IO;

var path = "./sonar.txt";


List<int> fileReader()
{
    List<int> returnList = new List<int>();
    foreach (string line in System.IO.File.ReadLines(path))
    {
        returnList.Add(Int16.Parse(line));
    }
    return returnList;
}

int tester()
{
    return 1;
}


int increasedCounter(List<int> list)
{
    int counter = 0;
    for (int i = 0; i < list.Count - 1; i++)
    {
        if (list[i + 1] > list[i])
        {
            counter++;
        }
    }
    return counter;
}

void partOne(List<int> sonar)
{
    Console.WriteLine("Part One:" + increasedCounter(sonar).ToString());
}

void partTwo(List<int> sonar)
{
    List<int> folds = new List<int>();
    for (int i = 0; i < sonar.Count - 2; i++)
    {
        folds.Add(sonar[i] + sonar[i + 1] + sonar[i + 2]);
    }
    Console.WriteLine("Part One:" + increasedCounter(folds).ToString());
}

void main()
{
    List<int> sonar = fileReader();
    partOne(sonar);
    partTwo(sonar);

}

main();


